pallets/assets-registry/src/lib.rs

```diff
@@ -137,13 +131,23 @@ pub mod pallet {
 	}
 
 	// Convert MultiLocation to a Chainbridge compatible resource id.
-	pub trait IntoResourceId {
+	pub trait IntoResourceId<T: Get<Option<u8>>> {
 		fn into_rid(self, chain_id: u8) -> [u8; 32];
 	}
 
-	impl IntoResourceId for MultiLocation {
+	impl<T: Get<Option<u8>>> IntoResourceId<T> for MultiLocation {
 		fn into_rid(self, chain_id: u8) -> [u8; 32] {
-			let mut rid = sp_io::hashing::blake2_256(&self.encode());
+			let mut rid = match T::get() {
+				Some(salt) => sp_io::hashing::blake2_256(
+					&self
+						.clone()
+						.pushed_with_interior(GeneralIndex(salt as u128))
+						// We have guaranteed length would never overflow when registering assets
+						.unwrap()
+						.encode(),
+				),
+				None => sp_io::hashing::blake2_256(&self.encode()),
+			};
```

Comment:
```suggestion
			let mut rid = sp_io::hashing::blake2_256(&match T::get() {
				Some(salt) => (&self, &salt),
				None => &self,
			});
```

Easier approach

Reply:
Oh, but we already have generated the rid for Phala PHA. Then we have to stick to your original design